django/core/management/commands/makemessages.py

```diff
@@ -443,9 +437,131 @@ def is_ignored(path, ignore_patterns):
                         locale_dir = self.default_locale_path
                     if not locale_dir:
                         locale_dir = NO_LOCALE_DIR
-                    all_files.append(TranslatableFile(dirpath, filename, locale_dir))
+                    all_files.append(self.translatable_file_class(dirpath, filename, locale_dir))
         return sorted(all_files)
 
+    def process_files(self, file_list):
+        u"""
+        Group translatable files by locale directory and run pot file build
+        process for each group.
+        """
+        file_groups = {}
+
+        for translatable in file_list:
+            file_group = file_groups.setdefault(translatable.locale_dir, [])
+            file_group.append(translatable)
+
+        for locale_dir, files in file_groups.items():
+            self.process_locale_dir(locale_dir, files)
+
+    def process_locale_dir(self, locale_dir, files):
+        u"""
+        Extract translatable literals from :param files:, creating or updating
```

Comment:
I'd suggest to remove the :param: stuff in the strings you've touched as that's not really something we've adopted throughout the code base.